### PR TITLE
Remove PR template

### PR DESCRIPTION
**Pull request by lenaschoenburg**
The template is verbose, doesn't apply to all teams, components and changes. It also polluted our git history because it was included in all merge commits.

I've extended the contributing guide a tiny bit to explain that referencing closed issues is important for changelog generation.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -7,48 +7,3 @@
...
-* [ ] I've separated structural from behavioural changes
--->
-
-## Definition of Done
-
-<!-- Please check the items that apply, before merging or (if possible) before requesting a review. -->
-
-_Not all items need to be done depending on the issue and the pull request._
-
-Code changes:
```
Comment (Zelldon) [inline .github/pull_request_template.md:32]: ❤️

**Review by ChrisKujawa — APPROVED** [review]
Awesomesauce 🚀 thanks

**Comment by oleschoenburg** [comment]
Hmmm, spotless / flexmark doesn't like that markdown and wants to reformat it to something broken:

```md
## Description

<!-- Please explain the changes you made here. -->

## Related issues

<!-- Which issues are closed by this PR or are related -->
close-->

```

**Comment by Zelldon** [comment]
@oleschoenburg btw should we backport this?

**Comment by oleschoenburg** [comment]
No backport needed, GH just takes the template from the default branch AFAIK

**Comment by Zelldon** [comment]
@oleschoenburg did you tried to add a newline after the comment?

**Pull request merged**